--- show.gsp
+++ src/templates/scaffolding/show.gsp	2009-02-25 07:09:30.000000000 +0000
@@ -32,7 +32,9 @@
                         props.each { p -> %>
                         <tr class="prop">
                             <td valign="top" class="name">${p.naturalName}:</td>
-                            <% if(p.isEnum()) { %>
+                            <% if(p.type in [org.joda.time.LocalTime, org.joda.time.LocalDate, org.joda.time.LocalDateTime, org.joda.time.DateTime ]) { %>
+							<td valign="top" class="value">\${fieldValue(bean:${propertyName}, field:'${p.name}')}</td>
+							<% } else if(p.isEnum()) { %>
                             <td valign="top" class="value">\${${propertyName}?.${p.name}?.encodeAsHTML()}</td>
                             <% } else if(p.oneToMany || p.manyToMany) { %>
                             <td  valign="top" style="text-align:left;" class="value">
